syntax = "proto3";

import "google/protobuf/empty.proto";

package order;

option java_multiple_files = true;
option java_package = "com.example.order";
option java_outer_classname = "OrderProto";

service OrderService {
  rpc ListOrders (google.protobuf.Empty) returns (ListOrdersResponse);
  rpc GetOrder (GetOrderRequest) returns (Order);
  rpc CreateOrder (CreateOrderRequest) returns (Order);
}

message Order {
  string id = 1;
  string product_id = 2;
  int32 quantity = 3;
  double total_price = 4;
}

message ListOrdersResponse {
  repeated Order orders = 1;
}

message GetOrderRequest {
  string id = 1;
}

message CreateOrderRequest {
  string product_id = 1;
  int32 quantity = 2;
}
